reactive/container_runtime_common.py

```diff
@@ -0,0 +1,53 @@
+from base64 import b64decode
+from subprocess import check_call
+
+from charms.layer import status
+from charms.reactive import (
+    clear_flag,
+    set_flag,
+    when,
+    when_not
+)
+
+from charmhelpers.core import hookenv, host
+
+
+@when_not('cgroups.modified')
+def enable_grub_cgroups():
+    """
+    Run script to enable cgroups
+    in GRUB.  Be aware, this will
+    reboot the host.
+
+    :return: None
+    """
+    cfg = hookenv.config()
+    if cfg.get('enable-cgroups'):
+        hookenv.log('Calling enable_grub_cgroups.sh and rebooting machine.')
+        check_call(['scripts/enable_grub_cgroups.sh'])
+        set_flag('cgroups.modified')
+
+
+@when('config.changed.custom-registry-ca')
+def install_custom_ca():
+    """
+    Installs a configured CA cert into the system-wide location.
+    """
+    ca_cert = hookenv.config().get('custom-registry-ca')
+    if ca_cert:
+        try:
+            # decode to bytes, as that's what install_ca_cert wants
+            _ca = b64decode(ca_cert)
+        except Exception:
+            status.blocked('Invalid base64 value for custom-registry-ca config')
+            return
+        else:
+            host.install_ca_cert(_ca, name='juju-custom-registry')
+            charm = hookenv.charm_name()
+            hookenv.log('Custom registry CA has been installed for {}'.format(charm))
+
+            # NB: containerd auto restarts when config changes. docker does not,
+            # so manage the appropriate layer flags to recycle dockerd.
```

Comment:
Is there a race condition with containerd? I'm worried that containerd's [config_changed](https://github.com/charmed-kubernetes/charm-containerd/blob/a77f8a2b26b93fdaa2af9d5fd0554fed9dd4232e/reactive/containerd.py#L328) handler might be called before install_custom_ca, not after.